usr/src/uts/armv8/os/ddi_impl.c

```diff
@@ -1157,21 +1157,49 @@ i_ddi_free_intr_phdl(ddi_intr_handle_impl_t *hdlp)
 	hdlp->ih_private = NULL;
 }
 
+
+dev_info_t *
+i_ddi_interrupt_parent(dev_info_t *pdip)
```

Comment:
The logic here seems at odds with `devicetree-specification-v0.4.pdf`.

The spec states: _Because the hierarchy of the nodes in the interrupt tree might not match the devicetree, the interrupt-parent property is available to make the definition of an interrupt parent explicit. The value is the phandle to the interrupt parent. **If this property is missing from a device, its interrupt parent is assumed to be its devicetree
parent**._

So perhaps we should just look for `interrupt-parent` and, if missing, return `ddi_get_parent` (or search upwards via that chain for a node with the `interrupt-controller` boolean property. We should probably also be searching with `DDI_PROP_DONTPASS`.

Also of interest is this snippet: _The root of the interrupt tree is determined when traversal of the interrupt tree reaches an interrupt controller node without an interrupts property and thus no explicit interrupt parent._